R-package/R/xgb.train.R

```diff
@@ -512,3 +381,435 @@ xgb.train <- function(params = list(), data, nrounds, evals = list(),
 
   return(bst)
 }
+
+# nolint start: line_length_linter
+#' @title XGBoost Parameters
+#' @description Convenience function to generate a list of named XGBoost parameters, which
+#' can be passed as argument `params` to [xgb.train()]. See the [online documentation](
+#' https://xgboost.readthedocs.io/en/stable/parameter.html) for more details.
+#'
+#' The purpose of this function is to enable IDE autocompletions and to provide in-package
+#' documentation for all the possible parameters that XGBoost accepts. The output from this
+#' function is just a regular R list containing the parameters that were set to non-default
+#' values. Note that this function will not perform any validation on the supplied arguments.
+#'
+#' If passing `NULL` for a given parameter (the default for all of them), then the default
+#' value for that parameter will be used. Default values are automatically determined by the
+#' XGBoost core library upon calls to [xgb.train()] or [xgb.cv()], and are subject to change
+#' over XGBoost library versions.
+#' @return A list with the entries that were passed non-NULL values. It is intended to
+#' be passed as argument `params` to [xgb.train()] or [xgb.cv()].
+#' @export
+#' @param objective (default=`"reg:squarederror"`)
+#' Specify the learning task and the corresponding learning objective or a custom objective function to be used.
+#'
+#' For custom objective, see [Custom Objective and Evaluation Metric](https://xgboost.readthedocs.io/en/latest/tutorials/custom_metric_obj.html)
+#' and [Custom objective and metric](https://xgboost.readthedocs.io/en/latest/tutorials/saving_model.html#custom-obj-metric) for more information,
+#' along with the end note for function signatures.
+#'
+#' Supported values are:
+#' - `"reg:squarederror"`: regression with squared loss.
+#' - `"reg:squaredlogerror"`: regression with squared log loss \eqn{\frac{1}{2}[log(pred + 1) - log(label + 1)]^2}.  All input labels are required to be greater than -1.  Also, see metric `rmsle` for possible issue  with this objective.
+#' - `"reg:logistic"`: logistic regression, output probability
+#' - `"reg:pseudohubererror"`: regression with Pseudo Huber loss, a twice differentiable alternative to absolute loss.
+#' - `"reg:absoluteerror"`: Regression with L1 error. When tree model is used, leaf value is refreshed after tree construction. If used in distributed training, the leaf value is calculated as the mean value from all workers, which is not guaranteed to be optimal.
+#'
+#'   Version added: 1.7.0
+#' - `"reg:quantileerror"`: Quantile loss, also known as "pinball loss". See later sections for its parameter and [Quantile Regression](https://xgboost.readthedocs.io/en/latest/python/examples/quantile_regression.html#sphx-glr-python-examples-quantile-regression-py) for a worked example.
```

Comment:
This is a Python example, do we want to include it here?

Reply:
I'd say python examples are still relevant.